### PR TITLE
docs(nms): Add contributing conventions for JavaScript

### PR DESCRIPTION
**Pull request by ato-dani**
Signed-off-by: Dani Menewuyelet <dmenewuy@fb.com>

## Summary
A contributing convention was added for Javascript developers. It currently only consists of 4 parts: General, Type Annotations, Documentation, and Linter. Also, a note was added on where to place js test files. 

## Test Plan
Run docusaurus locally also attached screenshots
Screen Shot 2021-07-20 at 10 48 13 AM
Screen Shot 2021-07-20 at 10 48 38 AM
. 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/contributing/contribute_conventions.md [hunk 1]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
     - Default to `.bash` except with specific reason
 - When a shell script passes around 100 lines, it's time to re-write it in Python or Go
 
+### Javascript
+
+**General**
+- Importing order should be: types, default components, and named components with each separated by a newline
+- Use `CamelCase` for component file names
```
Comment (andreilee) [inline docs/readmes/contributing/contribute_conventions.md]: Should be TitleCase

**Review by electronjoe — APPROVED** [review]
LGTM modulo addressing feedback

File: docs/readmes/contributing/contribute_conventions.md [hunk 2]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
...
+- Favour `map` and `forEach` functions in place of regular `for` loops
+- Refrain from using literal strings/numbers without defining them
+- Use strict equality(`===`) when comparing values
+
+**Type Annotations**
```
Comment (themarwhal) [inline docs/readmes/contributing/contribute_conventions.md]: I think for **Type Annotations** and **Linter** we might want to add a line to link the dev_testing instructions.

**Review by themarwhal — APPROVED** [review]
left one comment

File: docs/readmes/contributing/contribute_conventions.md (hunk 2 above)
Comment (ato-dani) [reply docs/readmes/contributing/contribute_conventions.md]: Yeah, that was what I think of too. So, I linked flow and eslint official website for now but will change them to link to dev_testing instruction when that pull request gets approved cause that makes more sense.

File: docs/readmes/contributing/contribute_conventions.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ Follow these conventions when making changes to the Magma codebase.
 - Unit tests should default to being placed in the same directory as the files they're testing, except for the following
     - Python: directly-adjacent `tests` directory
     - C/C++: directly-adjacent `tests` directory
+    - Javascript: directly-adjacent `__tests__` directory
```
Comment (hcgatewood) [inline docs/readmes/contributing/contribute_conventions.md]:
```suggestion
    - JavaScript: directly-adjacent `__tests__` directory
```

File: docs/readmes/contributing/contribute_conventions.md (hunk 2 above)
Comment (hcgatewood) [reply docs/readmes/contributing/contribute_conventions.md]:
```suggestion
**Type annotations**
```

File: docs/readmes/contributing/contribute_conventions.md [hunk 4]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
     - Default to `.bash` except with specific reason
 - When a shell script passes around 100 lines, it's time to re-write it in Python or Go
 
+### Javascript
+
+**General**
+- Importing order should be: types, default components, and named components with each separated by a newline
```
Comment (hcgatewood) [inline docs/readmes/contributing/contribute_conventions.md]:
```suggestion
- Import order should be: types, default components, and named components with each separated by a newline
```

File: docs/readmes/contributing/contribute_conventions.md [hunk 5]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
...
+- Use `TitleCase` for component file names
+- Favor `map` and `forEach` functions in place of regular `for` loops
+- Refrain from using literal strings/numbers without defining them
+- Use strict equality(`===`) when comparing values
+
+**Type Annotations**
+- All new code should be fully type anotatated
+- For mandatory type checks, we have a mandatory unit test that runs [flow](https://flow.org/) on the nms codebase
```
Comment (hcgatewood) [inline docs/readmes/contributing/contribute_conventions.md]:
```suggestion
- We have a mandatory unit test that runs the [flow](https://flow.org/) type checker on the NMS codebase
```

File: docs/readmes/contributing/contribute_conventions.md [hunk 6]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
...
+    - On CI, the check gets run as part of the `nms-flow-test` job
+
+**Documentation**
+- Document all the components and functions you created
```
Comment (hcgatewood) [inline docs/readmes/contributing/contribute_conventions.md]:
```suggestion
- Document all the components and functions you create
```

File: docs/readmes/contributing/contribute_conventions.md [hunk 7]
```diff
@@ -281,6 +282,31 @@ The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
...
+- Keep the docs up to date when you make changes
+- Use [JSDoc](https://jsdoc.app/index.html) tags to document your code
+- Use `//` to comment line of code, if it is not clear in what it is doing
+
+**Linter**
+- Run [eslint](https://eslint.org/) to lint your changes locally
```
Comment (hcgatewood) [inline docs/readmes/contributing/contribute_conventions.md]:
```suggestion
- Run [ESLint](https://eslint.org/) to lint your changes locally
```

**Review by hcgatewood — COMMENTED** [review]
Lgtm. Fix comments, and looks like you need to fix the DCO check

File: docs/readmes/contributing/contribute_conventions.md (hunk 2 above)
Comment (ato-dani) [reply docs/readmes/contributing/contribute_conventions.md]: @themarwhal I have now added the link to the dev_testing instructions since that got merged.

**Review by andreilee — APPROVED** [review]

**Pull request merged**